cmd/run.go

```diff
@@ -503,29 +505,55 @@ func init() {
 	runCmd.Flags().AddFlagSet(runCmdFlagSet())
 }
 
+func createFilesystems() map[string]afero.Fs {
+	// We want to eliminate disk access at runtime, so we set up a memory mapped cache that's
+	// written every time something is read from the real filesystem. This cache is then used for
+	// successive spawns to read from (they have no access to the real disk).
+	// Also initialize the same for `https` but the caching is handled manually in the loader package
+	osfs := afero.NewOsFs()
+	if runtime.GOOS == "windows" {
+		// This is done so that we can continue to use paths with /|"\" through the code but also to
+		// be easier to travers the cachedFs later as it doesn't work very well if you have windows
+		// volumes
+		osfs = fsext.NewTrimFilePathSeparatorFs(osfs)
+	}
+	return map[string]afero.Fs{
+		"file":  fsext.NewCacheOnReadFs(osfs, afero.NewMemMapFs(), 0),
+		"https": afero.NewMemMapFs(),
+	}
+}
+
 // Reads a source file from any supported destination.
-func readSource(src, pwd string, fs afero.Fs, stdin io.Reader) (*lib.SourceData, error) {
+func readSource(src, pwd string, filesystems map[string]afero.Fs, stdin io.Reader) (*loader.SourceData, error) {
 	if src == "-" {
 		data, err := ioutil.ReadAll(stdin)
 		if err != nil {
 			return nil, err
 		}
-		return &lib.SourceData{Filename: "-", Data: data}, nil
+		return &loader.SourceData{URL: &url.URL{Path: "-", Scheme: "file"}, Data: data}, nil
+	}
+	pwdURL := &url.URL{Scheme: "file", Path: filepath.ToSlash(filepath.Clean(pwd)) + "/"}
+	srcLocalPath := filepath.Join(pwd, src)
```

Comment:
Shouldn't there be a `filepath.IsAbs()` check here? It doesn't make sense to append absolute paths to the current working directory, even if I doubt it will be problematic.

Reply:
yeah, just found it while testing ...